test/ruby_lsp_rails/indexing_enhancement_test.rb

```diff
@@ -7,14 +7,12 @@ module RubyLsp
   module Rails
     class IndexingEnhancementTest < ActiveSupport::TestCase
       class << self
-        # For these tests, it's convenient to have the index fully populated with Rails information, but we don't have
-        # to reindex on every single example or that will be too slow
+        # For these tests, it's convenient to have the index fully populated with Rails information.
+        # Reindexing each time is slow, but prevents state leaking between tests.
         def populated_index
-          @index ||= begin
```

Comment:
@vinistock avoiding the memoization solves in test instability here. It's obviously slower though.

A few possible alternatives:
* Have each test be responsible to removing things it added to the index (maybe compare the state before and after in a `teardown` to ensure clean-up isn't forgotten)
* Have some naming convention for things we use in tests, and have them automatically removed from the index in a `teardown`.
* Have a way to skip indexing of RBS in `index_all` (which takes a large portion of the test time).

Reply:
How about we just add note to the test and use different names in each example? Trying to clean up all of the index state might be difficult and it will require exposing some internal APIs.